src/game/server/neo/neo_player.cpp

```diff
@@ -1215,70 +1270,79 @@ float CNEO_Player::GetFogObscuredRatio(CBaseEntity* target) const
 		return player->GetAbsVelocity().LengthSqr() > (runSpeedThreshold * runSpeedThreshold);
 		};
 
-	bool myPlayerIsMoving = isMoving(this); // Observer (me) is moving
 	bool targetIsMoving = isMoving(targetPlayer);
-	bool targetIsRunning = isRunning(targetPlayer);
-
-	// --- Apply Multipliers to Base Detection Chance ---
 
-	// Player Movement Impact
-	if (targetIsRunning) // Running is the most severe penalty
-	{
-		detectionChance *= MULT_TARGET_RUNNING;
-	}
-	else if (targetIsMoving) // Walking/strafing
+	// Class Impact:
+	// Assault class motion vision
+	if (GetClass() == NEO_CLASS_ASSAULT && targetIsMoving)
 	{
-		detectionChance *= MULT_TARGET_WALKING;
+		// I have motion vision
+		return 0.0f; // Not obscured
 	}
 
-	// Bot Movement Impact
-	if (myPlayerIsMoving)
+	if (GetClass() != NEO_CLASS_SUPPORT)
 	{
-		detectionChance *= MULT_MY_PLAYER_MOVING;
+		// thermal vision penalty against cloak
+		nSneakPenaltyCount += 5;
 	}
 
-	// Distance Impact
-	const Vector& myPos = GetAbsOrigin(); // TODO: May need GetBot()->GetPosition() equivalent
-	float currentRangeSq = (target->GetAbsOrigin() - myPos).LengthSqr(); // TODO: May need known.GetLastKnownPosition() equivalent
+	bool observerIsMoving  = isMoving(this); // Observer (me) is moving
+	bool observerIsRunning = isRunning(this);  // Observer (me) is running
```

Comment:
Could optimize with `observerIsMoving = observerIsRunning || isMoving(this)` so in the running case (which we have to calculate anyway), there's no need to call `isMoving` because running already implies moving.